src/settings.rs

```diff
@@ -355,35 +356,51 @@ impl Settings {
         settings
     }
 
-    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_CACHE_DIR
+    // Use or create the 'mcfly' folder in `$XDG_CACHE_DIR``
     pub fn mcfly_training_cache_path() -> PathBuf {
-        let cache_dir = Settings::mcfly_xdg_dir().cache_dir().to_path_buf();
-
-        Settings::mcfly_base_path(cache_dir).join(PathBuf::from("training-cache.v1.csv"))
+        Settings::xdg_base_strategy()
+            .cache_dir()
+            .join(PathBuf::from("mcfly/training-cache.v1.csv"))
     }
 
-    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_DATA_DIR
+    // Use or migrate to the 'mcfly' folder in `$XDG_STATE_DIR`
     pub fn mcfly_db_path() -> PathBuf {
-        let data_dir = Settings::mcfly_xdg_dir().data_dir().to_path_buf();
-
-        Settings::mcfly_base_path(data_dir).join(PathBuf::from("history.db"))
-    }
-
-    fn mcfly_xdg_dir() -> ProjectDirs {
-        ProjectDirs::from("", "", "McFly").unwrap()
+        let path = Settings::xdg_base_strategy()
```

Comment:
Is this backwards compatible?

Reply:
Sorry, forgot to migrate from data_dir, that's done now.